modules/aws-eks-self-managed-node-groups/locals.tf

```
@@ -49,16 +51,20 @@ locals {
     var.self_managed_ng
   )
 
+  # WINDOWS CONFIGURATION
+  compatible_with_windows_server_2022 = var.context.cluster_version >= "1.22" ? true : false 
   enable_windows_support = local.self_managed_node_group["launch_template_os"] == "windows"
+  windows_server_version = local.self_managed_node_group["windows_server_version"]
 
   predefined_ami_names = {
     amazonlinux2eks = "amazon-eks-node-${var.context.cluster_version}-*"
     bottlerocket    = "bottlerocket-aws-k8s-${var.context.cluster_version}-x86_64-*"
-    windows         = "Windows_Server-2019-English-Core-EKS_Optimized-${var.context.cluster_version}-*"
+    windows         = "Windows_Server-${local.windows_server_version}-English-Core-EKS_Optimized-${var.context.cluster_version}-*"
```

this is all that we need to change (plus add the variable to support):

```suggestion
    windows         = "Windows_Server-${var.windows_server_version}-English-Core-EKS_Optimized-${var.context.cluster_version}-*"
```

---

That variable doesn't exist **windows_server_version**, it only exists as a local. Since the **self_managed_ng** object only pass all the variables inside it, we are merging this variable that came from this object with the **default_self_managed_ng** from locals.

Clients can override these variable passing that to self_managed_ng object, that will be merged:
```
  self_managed_node_group = merge(
    local.default_self_managed_ng,
    var.self_managed_ng
  )
```

---

Correct, thats why I stated `plus add the variable to support` - we need to add this variable to allow users to specify which version they want to use - `2019` or `2022`, and `2019` should be the default since that is the current behavior

---

This variable should be passed inside the **self_managed_ng** object:
```
variable "self_managed_ng" {
  description = "Map of maps of `eks_self_managed_node_groups` to create"
  type        = any
  default     = {}
}
```

Since this object only forwards all the properties inside of that, i made the change in a way that the client can specify, but if doesn't, the default will be adopted, based on the cluster version (2022 for 1.22 and above or 2019 for below 1.22).

---

That works in the same way that **node_group_name** works for example, you can pass and override it, but if you don't, it will adopt a default value from locals:
```
node_group_name = "m4_on_demand"
```